src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -528,14 +529,16 @@ private async Task<Compilation> GetOrBuildDeclarationCompilationAsync(SolutionSe
                 ImmutableArray<SourceGeneratedDocumentState>? authoritativeGeneratedDocuments =
                     state.GeneratedDocumentsAreFinal ? state.GeneratedDocuments : null;
```

Comment:
It might not be bad to rename GeneratedDocumentsAreFinal to GeneratedDocumentsAreAuthoritative to preserve the naming convention a bit further.